lib/node_modules/@stdlib/random/base/mt19937/test/test.js

```diff
@@ -0,0 +1,211 @@
+var mt19937 = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.equal( typeof mt19937, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'attached to the main export is a method to generate normalized pseudorandom numbers', function test( t ) {
+	t.equal( typeof mt19937.normalized, 'function', 'has `normalized` method' );
+	t.end();
+});
+
+tape( 'attached to the main export is a method to generate linear congruential pseudorandom number generator', function test( t ) {
+	t.equal( typeof mt19937.factory, 'function', 'has `factory` method' );
+	t.end();
+});
+
+tape( 'attached to the main export is the generator name', function test( t ) {
+	t.equal( mt19937.NAME, 'mt19937', 'has `NAME` property' );
+	t.end();
+});
+
+tape( 'attached to the main export is the generator seed', function test( t ) {
+	t.equal( typeof mt19937.SEED, 'number', 'has `SEED` property' );
+	t.end();
+});
+
+tape( 'attached to the main export is the minimum possible generated number', function test( t ) {
+	t.equal( mt19937.MIN, 1, 'has `MIN` property' );
+	t.end();
+});
+
+tape( 'attached to the main export is the maximum possible generated number', function test( t ) {
+	t.equal( mt19937.MAX, UINT32_MAX, 'has `MAX` property' );
+	t.end();
+});
+
+tape( 'the function returns pseudorandom integers strictly between 0 and 2^32-1 (inclusive)', function test( t ) {
+	var v;
+	var i;
+	for ( i = 0; i < 1e4; i++ ) {
+		v = mt19937();
+		t.equal( typeof v, 'number', 'returns a number' );
+		t.equal( isPositiveInteger( v ), true, 'returns a positive integer' );
+		t.equal( v >= 1 && v <= UINT32_MAX, true, 'returns an integer between 1 and 2^32-1 (inclusive)' );
+	}
+	t.end();
+});
+
+tape( 'the `normalized` method returns pseudorandom numbers strictly between 0 (inclusive) and 1 (exclusive)', function test( t ) {
+	var v;
+	var i;
+	for ( i = 0; i < 1e4; i++ ) {
+		v = mt19937.normalized();
+		t.equal( typeof v, 'number', 'returns a number' );
+		t.equal( v >= 0.0 && v < 1.0, true, 'returns a number between 0 (inclusive) and 1 (exclusive)' );
+	}
+	t.end();
+});
+
+tape( 'attached to the `normalized` method is the generator name', function test( t ) {
+	t.equal( mt19937.normalized.NAME, 'mt19937', 'has `NAME` property' );
+	t.end();
+});
+
+tape( 'attached to the `normalized` method is the generator seed', function test( t ) {
+	t.equal( typeof mt19937.normalized.SEED, 'number', 'has `SEED` property' );
+	t.end();
+});
+
+tape( 'attached to the `normalized` method is the minimum possible generated number', function test( t ) {
+	t.equal( mt19937.normalized.MIN, 0.0, 'has `MIN` property' );
+	t.end();
+});
+
+tape( 'attached to the `normalized` method is the maximum possible generated number', function test( t ) {
+	t.equal( mt19937.normalized.MAX, ((UINT32_MAX-1.0))/(UINT32_MAX), 'has `MAX` property' );
```

Comment:
I don't believe this is correct. We now use 53-bits of precision, not the 32 bits of precision algo.